### PR TITLE
Simple final approach for MC

### LINKED ISSUE #3894: MPC Final approach land speed
Implement parameter to set final approach altitude and land speed.
This would allow slower descend rate for the final x meters.

@LorenzMeier I propose this with a default value of 1.5m above home and around 60% land speed. Any thoughts?

### PR DESCRIPTION
**Pull request by AndreasAntener**
Credits to @sanderux for the initial work, this is the impl I would prefer.
@dagar please check if this is to your liking
Fixes #3894

### CODE REVIEW

**Comment by dagar** [comment]
MPC_FNA_SPD seems unnecessary. Why not let it use the regular values for waypoint navigation and then MPC_LAND_SPEED below MPC_FNA_ALT?

Do we need to specify use terrain for landing (MPC_FNA_USE_TER)? Wouldn't it make sense to respect the existing MPC_ALT_MODE?

**Comment by sanderux** [comment]
Disregarding @dagar's comments that i agree with, doesnt this: https://github.com/PX4/Firmware/pull/5828/files#diff-fc77c3ef569029d45764664c75d4b0c1R1587 make the param MPC_FNA_USE_TER more of a 'Force terrain' considering it would also use terrain if available when the param is off?

**Comment by dagar** [comment]
Personally I don't like having so many params for every little thing. I'd rather specify to use terrain data by sending the LAND waypoint with MAV_FRAME_GLOBAL_TERRAIN_ALT instead of MAV_FRAME_GLOBAL_RELATIVE_ALT, but that's more work.

**Comment by dagar** [comment]
Here's the even simpler version I've been using.
https://github.com/dagar/Firmware/commit/6db7a639e315e993793663ef5daca9a68a14528f

**Comment by LorenzMeier** [comment]
Folks, do we discuss this on the dev call? It sounds it needs touching base first?

**Comment by AndreasAntener** [comment]
Probably a good idea :) But we need to start with how a landing should work / be commanded from a user perspective. This here is only one part of the technical solution.

File: src/modules/mc_pos_control/mc_pos_control_params.c [hunk 1]
```diff
@@ -360,6 +360,36 @@ PARAM_DEFINE_FLOAT(MPC_TILTMAX_LND, 12.0f);
...
+ * Final approach altitude
+ *
+ * Slow down landing descent to final approach speed if below final approach altitude.
+ *
+ * @unit meters
+ * @min 0.0
+ * @decimal 2
+ * @group Multicopter Position Control
+ */
+PARAM_DEFINE_FLOAT(MPC_FNA_ALT, 0.0f);
```
Comment (julianoes) [inline src/modules/mc_pos_control/mc_pos_control_params.c]: Shouldn't this number be > 0 ?

**Comment by sanderux** [comment]
I vote we merge this until a better solution has been found.
@AndreasAntener ok if i rebase this?

I will flight test this asap and review based on that.

**Comment by AndreasAntener** [comment]
Fine with me

**Comment by dagar** [comment]
Do we need MPC_FNA_SPD (final approach speed)? Isn't it sufficient to respect the regular Z limit until the switch to landing speed??

**Comment by sanderux** [comment]
I think people will have a hard time understanding the landing sequence if we do

**Comment by dagar** [comment]
Oh, I had it backwards.

What I was thinking of was just using the regular controller Z velocity until the final approach, and then use landing speed for the soft touchdown.

**Comment by LorenzMeier** [comment]
We hacked this strategy on master in position and user controlled modes. I will make sure we bring the equivalent over from this PR.

I think what confused me was the fancy title of this PR. I never got what it was doing, so we ended up doing parallel work. We'll take the hit of merging the two parallel efforts to not burden with Ada doing that. Sorry for the communication mishap.

**Comment by TSC21** [comment]
@LorenzMeier I assume this PR lacks an update?

**Comment by LorenzMeier** [comment]
The changes from @Stifael contain a version of this, so I'm closing it.

**Pull request closed without merging**